### PR TITLE
Improve handling for term/year features

### PR DESCRIPTION
**Pull request by bdewilde**
## changes

- removed hacky "num terms checkin" logic from student-term dataset functionality, and replaced it with a more general solution: a "cleanup" function applied to labeled data that dynamically nulls out values coming from years in the future (after the student-term used for prediction) to prevent data leakage
- as a result, included a few extra features that were previously dropped, including the fraction of credits earned year-by-year
- in order for new logic to work, renamed a few features to have a consistent "*_year_X" naming scheme
- also, added a parser to the base pdp cohort dataset schema that nullifies 0 values for "first year to *" fields, which are now included as features
- finally, fixed a bug in a couple CIP-based features that resulted in them being interpreted as numeric values rather than categorical (**note:** this is a minor breaking change)

## context

Still struggling to train a model with performance I'd consider sufficient. Truly grasping at straws. Got any more features to try? I'm all ears!

Also, re: schema change: https://datakind.slack.com/archives/C07KBDVCFUZ/p1729642107713779

## questions

- The logic for nullifying "future" values is a bit tricky. The corresponding unit test makes me more confident it's doing what I want, but please look carefully at this and convince yourself: Is this logic correct?

### CODE REVIEW

File: src/student_success_tool/analysis/pdp/dataops.py [hunk 1]
```diff
@@ -198,6 +161,84 @@ def standardize_course_dataset(df: pd.DataFrame) -> pd.DataFrame:
...
+    """
+    return (
+        df.pipe(
+            drop_columns_safely,
+            cols=[
+                # metadata
+                "institution_id",
+                "term_id",
+                "academic_year",
+                "academic_term",
```
Comment (skvorekn) [inline src/student_success_tool/analysis/pdp/dataops.py:177]: Adding some hypotheses in favor of keeping some of these variables, but I don't feel strongly!

Hypothesis around seasonal patterns: students enrolled in the summer look different than other terms

File: src/student_success_tool/analysis/pdp/dataops.py [hunk 2]
```diff
@@ -198,6 +161,84 @@ def standardize_course_dataset(df: pd.DataFrame) -> pd.DataFrame:
...
+                "cohort",
+                "cohort_term",
```
Comment (skvorekn) [inline src/student_success_tool/analysis/pdp/dataops.py:179]: Hypothesis: students that first enroll in the Fall are different than students that first enroll in the Spring/Summer

**Review by skvorekn — APPROVED** [review]
I like it! Pending our assumptions about the PDP column definitions are correct, I feel good about this. I like that you did >= rather than > just to be safe

File: src/student_success_tool/analysis/pdp/dataops.py (hunk 1 above)
Comment (bdewilde) [reply src/student_success_tool/analysis/pdp/dataops.py:177]: I don't feel strongly, either. Let's err on the side of inclusion -- if the model finds it useful, then so be it.

File: src/student_success_tool/analysis/pdp/dataops.py (hunk 2 above)
Comment (bdewilde) [reply src/student_success_tool/analysis/pdp/dataops.py:179]: Same here.

**Pull request merged**